### DISCUSSION TITLE
Verification of JWT token on a third-party application

### DISCUSSION THREAD

**Discussion by Scylin232** [pocketbase/pocketbase#1630] [Q&A] [answered]
Hi, I'm trying to figure out how to validate PocketBase tokens in my ASP.NET Core application, usually people use a secret key for this, but our favorite gopher generates it dynamically based on common and user data. What are the theoretical solutions?

**Comment by ganigeorgiev** [comment, marked as answer]
You can call the [refresh endpoint](https://pocketbase.io/docs/api-records/#auth-record-actions) and it will "validate" the token for you:

```
curl -X POST http://127.0.0.1:8090/api/collections/users/auth-refresh -H 'Authorization: YOUR_TOKEN'
```

The above will fail if the token is invalid or expired.

**Reply by Scylin232** [reply]
But the user will lose access to the current session, right? If yes, is there a way out of this?

**Reply by ganigeorgiev** [reply]
The refresh endpoint doesn't invalidate old tokens. It just generate a new auth token and return the most recent auth record data. Note that PocketBase is completely stateless and doesn't really have sessions, since tokens are not stored on the server.

**Reply by Scylin232** [reply]
Thanks!
